runbotics/runbotics-orchestrator-ui/src/src-app/components/cron/fields/Minutes/Minutes.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, useMemo } from 'react';
+
+import CustomSelect from '../../components/CustomSelect';
+import { UNITS } from '../../constants';
+import DEFAULT_LOCALE_EN from '../../locale';
+import { PeriodTypes } from '../../types';
+import { classNames } from '../../utils';
+import { MinutesProps } from './Minutes.types';
+
+// eslint-disable-next-line complexity
+const Minutes: FC<MinutesProps> = ({ 
+    value, 
+    setValue, 
+    locale, 
+    className, 
+    disabled, 
+    readOnly, 
+    leadingZero, 
+    clockFormat, 
+    period 
+}) => {
+    const internalClassName = useMemo(
+        () =>
+            classNames({
+                'react-js-cron-field': true,
+                'react-js-cron-minutes': true,
+                [`${className}-field`]: !!className,
+                [`${className}-minutes`]: !!className,
+            }),
+        [className],
+    );
+
+    const fieldPrefix = useMemo(
+        () =>
+            period === PeriodTypes.HOUR 
+                ? (
+                    <span>
+                        {locale.prefixMinutesForHourPeriod || DEFAULT_LOCALE_EN.prefixMinutesForHourPeriod}
+                    </span>
+                ) 
+                : (
+                    <span>
+                        {locale.prefixMinutes || DEFAULT_LOCALE_EN.prefixMinutes}
+                    </span>
+                ),
+        [period, locale.prefixMinutesForHourPeriod, locale.prefixMinutes]
+    );
+
+    const fieldSuffix = useMemo(
```

Comment:
imo useMemo should be used in cases like long and complicated calculations going on or mapping a massive array in it. I don't see any of those here so I'd suggest using `useState`+`useEffect`

Reply:
I used component PeriodDefinition instead.
thanks for pointing this out

Reply:
done